### PR TITLE
Changing 'land' to 'snow' to better reflect analysis jobs

### PR DESCRIPTION
**Pull request by CoryMartin-NOAA**
This PR changes 'land' everywhere to 'snow' to differentiate that this is snow DA and not land DA. There will need to be a corresponding G-W PR to do this in the workflow.

### CODE REVIEW

File: ush/ufsda/yamltools.py
Comment (YoulongXia-NOAA) [inline ush/ufsda/yamltools.py:1]: snow -> SNOW?

**Review by RussTreadon-NOAA — DISMISSED** [review]
Looks reasonable to me.  

Approve pending success from CI tests

File: ush/ufsda/yamltools.py
Comment (YoulongXia-NOAA) [inline ush/ufsda/yamltools.py]: snow_BKG_YYYYmmddHHMMSS -> SNOW_BKG_YYYYmmddHHMMSS, as LAND is capital, I am not sure. The same case is for snow_BKG_ISOTIME -> SNOW_BKG_ISOTIME.

Comment (YoulongXia-NOAA) [reply ush/ufsda/yamltools.py]: Others looks fine with me.

**Review by YoulongXia-NOAA — DISMISSED** [review]
If you think lower case snow is fine, it looks good to me.

File: ush/ufsda/yamltools.py
Comment (CoryMartin-NOAA) [reply ush/ufsda/yamltools.py]: Thanks @YoulongXia-NOAA for catching my mistake!

**Comment by jiaruidong2017** [comment]
If we consider the soil DA is added in the future, I would suggest to keep land directories at: `${PROJECT_SOURCE_DIR}/test/land/` and `${PROJECT_SOURCE_DIR}/ush/land/`.

File: parm/snow/letkfoi/letkfoi.yaml
Comment (yuanxue2870) [inline parm/snow/letkfoi/letkfoi.yaml:57]: I assume we will introduce "soilinc" later, correct?

Comment (CoryMartin-NOAA) [reply parm/snow/letkfoi/letkfoi.yaml:57]: My logic here, and it is up for discussion, is that soil DA is strongly coupled with the atmosphere, so it would need a more holistic solution when that is ready in JEDI. This snow DA is a separate analysis (for now), hence why everything is defined here as 'snow' specifically

**Comment by jiaruidong2017** [comment]
@CoryMartin-NOAA Need to update `sorc/land-jediincr/sorc/apply_incr_noahmp_snow.f90` by changing from `landinc` to `snowinc`.

File: parm/snow/letkfoi/letkfoi.yaml
Comment (yuanxue2870) [reply parm/snow/letkfoi/letkfoi.yaml:57]: Got it. Thanks.

**Comment by CoryMartin-NOAA** [comment]
Thanks @jiaruidong2017 , I was trying to debug but that is the problem.

**Comment by CoryMartin-NOAA** [comment]
Before making the changes to the Fortran increment code, I want to make sure @ClaraDraper-NOAA @barlage and others are okay with the name change from `land` to `snow`

**Comment by jiaruidong2017** [comment]
I am fine with the name change from `landinc` to `snowinc`, because it is the snow specific in `apply_incr_noahmp_snow.f90`.

**Comment by barlage** [comment]
Fine with me, I'm always in favor of more descriptive naming.

**Review by ClaraDraper-NOAA — DISMISSED** [review]
looks good to me @CoryMartin-NOAA. It's kinda annoying to have to do, so thanks for addressing this now. I think I labelled everything in my global_workflow PR as "gsi_soil", but I will go back and double-check this.

File: parm/snow/letkfoi/letkfoi.yaml
Comment (ClaraDraper-NOAA) [reply parm/snow/letkfoi/letkfoi.yaml:57]: I agree with Cory. If all goes to plan, the soil DA from JEDI will be strongly coupled, so there will not be a separate increment file. Until we have a single land DA update, which is 5-10 years away, having clearly named files will be really helpful.

**Comment by CoryMartin-NOAA** [comment]
Thanks all, let me make the change in the fortran code, issue a PR there, then update the hash as part of this PR once that is merged.

**Review by ClaraDraper-NOAA — APPROVED** [review]

**Review by YoulongXia-NOAA — APPROVED** [review]
I looked at the change. It is good to me. I approve this PR.

**Review by jiaruidong2017 — APPROVED** [review]
Approve pending on success from CI test

**Review by yuanxue2870 — APPROVED** [review]

**Comment by jiaruidong2017** [comment]
The four failed tests seem caused by trying to submit batch jobs.

**Comment by CoryMartin-NOAA** [comment]
The failed tests are known to fail outside of this PR, I will merge. Thanks!

**Pull request merged**
